Adding handling of more use cases for namespace naming

ZdravkoD opened the pull request:
Example use cases:
1) anonymous namespace
2) 'o2', 'o2codecheck', ...

sawenzel commented:
This is a good improvement. Can you change the commit message to mention "namespace" somewhere. Moreover, I find the fix suggestion from "TPC" to "t_p_c" unnatural. We should not introduce "_" automatically (or only between small and capital letters).

ZdravkoD commented:
I've rebased the branch on origin/master and added a check for use-case with consecutive capital letters, if we have such case we don't make any automatic corrections.
